read-fonts/src/tables/layout/script.rs

```diff
@@ -67,6 +80,51 @@ impl<'a> ScriptList<'a> {
     }
 }
 
+impl<'a> Script<'a> {
+    /// If the script contains a language system with the given tag, returns
+    /// the index.
+    pub fn lang_sys_index_for_tag(&self, tag: Tag) -> Option<u16> {
+        self.lang_sys_records()
+            .binary_search_by_key(&tag, |rec| rec.lang_sys_tag())
+            .map(|index| index as u16)
+            .ok()
+    }
+
+    /// Returns the language system with the given index.
+    pub fn lang_sys(&self, index: u16) -> Result<TaggedElement<LangSys<'a>>, ReadError> {
+        self.lang_sys_records()
+            .get(index as usize)
+            .ok_or(ReadError::OutOfBounds)
+            .and_then(|rec| {
+                Ok(TaggedElement::new(
+                    rec.lang_sys_tag(),
+                    rec.lang_sys(self.offset_data())?,
+                ))
+            })
+    }
+}
+
+impl<'a> LangSys<'a> {
+    /// If the language system references a feature with the given tag,
+    /// returns the index of that feature in the specified feature list.
+    ///
+    /// The feature list can be obtained from the `feature_list` method on
+    /// the parent [Gsub](crate::tables::gsub::Gsub) or
+    /// [Gpos](crate::tables::gpos::Gpos) tables.
+    pub fn feature_index_for_tag(&self, list: &FeatureList, tag: Tag) -> Option<u16> {
+        let records = list.feature_records();
+        for feature_ix in self.feature_indices() {
+            let feature_ix = feature_ix.get();
+            if let Some(rec) = records.get(feature_ix as usize) {
+                if rec.feature_tag() == tag {
+                    return Some(feature_ix);
+                }
+            }
+        }
+        None
```

Comment:
could also just be,

```rust
self.feature_indices().map(|ix| ix.get())
    .find(|feature_ix| records.get(feature_ix as usize).map(|rec| rec.feature_tag()) == Some(tag))
```

but I'm just having fun

Reply:
snazzy... taken with a sprinkle of `.iter()` and `&`